client/sites-dashboard/components/sites-site-plan.tsx

```diff
@@ -0,0 +1,59 @@
+import styled from '@emotion/styled';
+import { useSelector } from 'react-redux';
+import JetpackLogo from 'calypso/components/jetpack-logo';
+import { getSitePlan } from 'calypso/state/sites/selectors';
+import { isNotAtomicJetpack } from '../utils';
+import { PlanRenewNag } from './sites-plan-renew-nag';
+import type { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+
+const SitePlanContainer = styled.div`
+	display: inline;
+	> * {
+		vertical-align: middle;
+		line-height: normal;
+	}
+`;
+
+const SitePlanIcon = styled.div`
+	display: inline-block;
+	margin-inline-end: 6px;
+`;
+
+const PlanRenewNagContainer = styled.div`
+	line-height: 20px;
+`;
+
+interface SitePlanProps {
+	site: SiteExcerptData;
+	userId: number | null;
+}
+
+export const SitePlan = ( { site, userId }: SitePlanProps ) => {
+	const plan = useSelector( ( state ) => {
+		if ( site?.options?.wpcom_production_blog_id ) {
+			return getSitePlan( state, site?.options?.wpcom_production_blog_id );
+		}
+		return site.plan;
+	} );
+
+	return (
+		<SitePlanContainer>
+			{ isNotAtomicJetpack( site ) && ! plan?.expired && (
+				<SitePlanIcon>
+					<JetpackLogo size={ 16 } />
+				</SitePlanIcon>
+			) }
+			{ site.plan?.expired ? (
```

Comment:
We could display nag pointing to the parent site, however, this wouldn't make sense if the user doesn't have access to the parent site. Maybe we should also add an explicit `! is_ staging` check here?

Reply:
I ended up changing the function a little bit. I only use the label instead of the plan. I also used is staging to determine if the nag will be shown or not.